graph/push.go

```diff
@@ -25,6 +24,14 @@ import (
 
 var ErrV2RegistryUnavailable = errors.New("error v2 registry unavailable")
 
+type ImagePushConfig struct {
```

Comment:
I think maybe it makes sense to structure what's written to the out stream to api/types as you said